### PR TITLE
fix(curriculum): use regex to avoid differences in how browsers calculate style property values for Decrease Opacity of an Element challenge

### LINKED ISSUE #37856: Decrease the Opacity of an Element - Tests won't pass on Safari after removing backticks
**Describe your problem and how to reproduce it:**

Noticed the issue in [this forum post](https://www.freecodecamp.org/forum/t/will-not-pass-test-but-code-is-typed-in-identical-to-video/329446). User had backticks around their code originally - 
```css
 `opacity: 0.7;`
```
but even after I removed them, the tests won't pass on Safari. 

Screen Shot 2019-11-29 at 11 44 42 PM

It works on Chrome though.

**Add a Link to the page with the problem:**

https://www.freecodecamp.org/learn/responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element

**Tell us about your browser and operating system:**
* Browser Name: Safari 
* Browser Version: 13.0.3
* Operating System: macOS Catalina

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

The [Decrease Opacity of an Element](https://www.freecodecamp.org/learn/responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element) challenge tests have been changed before to address this same issue, but they do not work on all browsers (like Safari).  For example, the original test used the `getComputedStyle` method and would return values like 0.699999887 for opacity.  

This PR uses regex to check the opacity value. to protect against differences in how browsers calculate the styling of elements.

Closes #37856

### CODE REVIEW

**Review by moT01 — COMMENTED** [review]
I'm not sure I'm a fan of replacing the spaces - It allows for something like `opa city: .7` to pass.

**Comment by RandellDawson** [comment]
> I'm not sure I'm a fan of replacing the spaces - It allows for something like opa city: .7 to pass.

I think that is a very small edge case, but if you insist, would you be OK with the following that still tries to keep the code simple?
```js
const styles = $('style').text(); assert(/\.links{\S*opacity:0?\.70*(;\S*}|})/.test(styles.replace(/\s/g ,'') && /opacity/.test(styles));
```

**Comment by moT01** [comment]
I'm just giving my opinion - I would prefer proper functionality over cleaner code - or at least as close as we can get. I don't think the regex is that complicated. Your first example only checks for a space in the word `opacity`, so a space in `links` or `0.7` won't get caught. But, if you want to leave it alone - we can let someone else weigh in.

Oh, also - on one of the other PR's - I noticed that capitalization is still valid CSS, so we should maybe have the `i` flag on regex's to ignore case - this is probably more of an issue since someone could easily have the caps lock on or accidentally start with an uppercase letter.

**Comment by RandellDawson** [comment]
@moT01 So you want me to go with the second assert I suggested above?

> Oh, also - on one of the other PR's - I noticed that capitalization is still valid CSS, so we should maybe have the i flag on regex's to ignore case - this is probably more of an issue since someone could easily have the caps lock on or accidentally start with an uppercase letter.

Not sure what PR you are referring to here.  If I need to fix another PR, please add the comment to it.

Thanks.

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ Set the <code>opacity</code> of the anchor tags to 0.7 using <code>links</code>
 ```yml
 tests:
   - text: Your code should set the <code>opacity</code> property to 0.7 on the anchor tags by selecting the class of <code>links</code>.
-    testString: assert(getComputedStyle($('.links')[0]).opacity == '0.7');
+    testString: assert(/\.links{\S*opacity:0?\.70*(;\S*}|})/.test($('style').text().replace(/\s/g ,'')));
```
Comment (thecodingaviator) [inline curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md]:
```suggestion
    testString: assert(/\.links{\S*opacity:\s?0?\.70*(;\S*}|})/.test($('style').text().replace(/\s/g ,'')));
```

Comment (thecodingaviator) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md]: How does this look?

**Review by Manish-Giri — APPROVED** [review]
LGTM. Works on Safari 💯

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md (hunk 1 above)
Comment (RandellDawson) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md]: @thecodingaviator No, if anything it would need to account for something like:
```
.links {
  opacity: .7
}
```

The assert below would be more flexible:
```
assert(/\.links{\s*opacity\s*:\s*0?\.70*\s*(;[\s\S]*}|\s*})/.test($('style').text()));
```

Comment (thecodingaviator) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/decrease-the-opacity-of-an-element.english.md]: We need to account for things like this:

image

image

**Comment by RandellDawson** [comment]
@thecodingaviator @Manish-Giri @moT01 I have updated the regex in hopes of improving what I had the first time around.  Please let me know if this is merge worthy.

Thank you.

**Review by thecodingaviator — APPROVED** [review]
LGTM now

**Comment by thecodingaviator** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝
